### ISSUE TITLE
Search: Parent/Child Results

### ISSUE THREAD

**Issue by orangewolf** [scientist-softserv/louisville-hyku#53]
# Summary
Ability to search for children's metadata and OCR within the parent's uv

# Acceptance Criteria

* [x] User should be able to search the catalog for metadata and OCR and get results (via its Parents only)
* [x] When searching for OCR on a parent work, the children results should get displayed and highlighted in its UV, if there is a match
  - ~~Will it be possible to then click on the child to be redirected to the child's show page?~~
* [x] doing a parent uv search for text from the `searchable_text` field of it's child(ren) should highlight the child(ren) thumbnail(s) that contained the searched for term
* [x] the "more information" pane on the right side of the parent uv viewer should contain the complete text of the `searchable_text` field
  - ref: #53 
  - example: https://collections.library.yale.edu/catalog/15479583

# Screenshots or Video

# Testing Instructions
Url: TBD
* TBD

**Comment by orangewolf** [comment]
Linked Items
---
[[5] Search](https://gitlab.com/notch8/louisville-hyku/-/issues/97)

[[5] Handle existing OCR/transcripts instead of generating new ones](https://gitlab.com/notch8/louisville-hyku/-/issues/20)

[Render Child Works in the UV](https://gitlab.com/notch8/louisville-hyku/-/issues/53)

[catalog searching for child metadata should return parents](https://gitlab.com/notch8/louisville-hyku/-/issues/134)

[remove faceted links for children](https://gitlab.com/notch8/louisville-hyku/-/issues/141)

**Comment by orangewolf** [comment]
Related Merge Requests
---
[i97 - search returns parents only](https://gitlab.com/notch8/louisville-hyku/-/merge_requests/80)

[100 - Iiif search parents](https://gitlab.com/notch8/louisville-hyku/-/merge_requests/103)

[UV child level metadata and customizations](https://gitlab.com/notch8/louisville-hyku/-/merge_requests/151)
!151

[142 - update facet link paths if record is a child](https://gitlab.com/notch8/louisville-hyku/-/merge_requests/152)

**Comment by orangewolf** [comment]
Alisha May 2022
---
searching for ocr from a parent work is functional on staging.

@rachel.howard @rtkuehn in order to get this working on your dev server with the updated code I just pushed to you, please do the following

``` bash
# from the rails console
- switch into the account # switch!(Account.first)
- run the code under "Reindex Everything For Large Systems" at https://playbook-staging.notch8.com/en/samvera/hyrax/how_to/reindex_work

# from the admin dashboard
- settings >> account
- check the box that says "ssl configured"
- save
```

**Comment by orangewolf** [comment]
Randy May 2022
---
searching for ocr from a parent work is functional on staging.

@rachel.howard @rtkuehn in order to get this working on your dev server with the updated code I just pushed to you, please do the following

``` bash
# from the rails console
- switch into the account # switch!(Account.first)
- run the code under "Reindex Everything For Large Systems" at https://playbook-staging.notch8.com/en/samvera/hyrax/how_to/reindex_work

# from the admin dashboard
- settings >> account
- check the box that says "ssl configured"
- save
```

**Comment by orangewolf** [comment]
Rachel May 2022
---
searching for ocr from a parent work is functional on staging.

@rachel.howard @rtkuehn in order to get this working on your dev server with the updated code I just pushed to you, please do the following

``` bash
# from the rails console
- switch into the account # switch!(Account.first)
- run the code under "Reindex Everything For Large Systems" at https://playbook-staging.notch8.com/en/samvera/hyrax/how_to/reindex_work

# from the admin dashboard
- settings >> account
- check the box that says "ssl configured"
- save
```

**Comment by orangewolf** [comment]
Randy May 2022
---
When I ran the large method, it really wasn't clear if it succeeded. So, I assumed it failed and ran the simple method. It failed with activefedora objectnotfound errors and stopped reindexing multiple times. However, I was able to use the url file created by the large method to look up the complete urls for each activefedora objectnotfound error that popped up and manually delete them from fedora. Problem fixed! The simple reindex finished successfully a short time ago.

**Comment by orangewolf** [comment]
Alisha May 2022
---
understood! the large method doesn't exactly give a "succeeded" method...it just doesn't fail. :sweat_smile:

glad you got it all figured out though!

**Comment by orangewolf** [comment]
Alisha May 2022
---
this ticket is related to #20 as well. #20 is for the ability to search metadata (e.g. the `searchable_text` field) of a child work, from the _child_ uv. this later created ticket includes being able to search the metadata of a child work, from the _parent_ uv.

completing this ticket should suffice #20.

please specifically refer to [this thread](https://gitlab.com/notch8/louisville-hyku/-/issues/20#note_945149378) on that ticket for details and conversation.

TLDR;
- [ ] doing a catalog search for text from the `searchable_text` field of a child should return the parent
- [ ] clicking on that parent should show the uv on the parent show page, and the child(ren) thumbnail(s) that contained the searched for term in its `searchable_text` field should be highlighted
- [ ] the "more information" pane on the right side of the uv viewer should contain the complete `searchable_text` field

**Comment by orangewolf** [comment]
Alisha May 2022
---
this ticket is related to #20 as well. #20 is for the ability to search metadata (e.g. the `searchable_text` field) of a child work, from the _child_ uv. this later created ticket includes being able to search the metadata of a child work, from the _parent_ uv.

completing this ticket should suffice #20.

please specifically refer to [this thread](https://gitlab.com/notch8/louisville-hyku/-/issues/20#note_945149378) on that ticket for details and conversation.

TLDR;
- [ ] doing a catalog search for text from the `searchable_text` field of a child should return the parent
- [ ] clicking on that parent should show the uv on the parent show page, and the child(ren) thumbnail(s) that contained the searched for term in its `searchable_text` field should be highlighted
- [ ] the "more information" pane on the right side of the uv viewer should contain the complete `searchable_text` field

**Comment by orangewolf** [comment]
Rachel May 2022
---
@alishaevn I do like the blue search icon. However, here's what I'm seeing in the viewer for the first search result. I'm not sure what's happening.



![Image](https://user-images.githubusercontent.com/1054448/203325172-20774a5b-f9ac-43e6-860c-0c1f6796dce3.png)

**Comment by orangewolf** [comment]
Alisha May 2022
---
re: the blue search icon being suffice - great!

what's happening there is the "zoom" feature on the uv. when there's an ocr search result, that term would be zoomed in on. however, for the metadata there isn't a term to zoom in on. so the zoom appears to shoot to the middle of the two pages. we've removed the zoom from the uv config for this reason. (also because some may find the zoom disorienting).

the new code pushed to randy yesterday should have contained this change.

**Comment by orangewolf** [comment]
Kirk September 2022
---

Searchable text now is in the "more information" pane


![Image](https://user-images.githubusercontent.com/1054448/203325471-8953a9cd-6a7a-474b-a825-573110de9f1c.png)



When searching for a word from the searchable_text field, the blue highlight should still show up

searched for "GANG"
https://lv-hyku-staging.notch8.cloud/concern/texts/a30fca72-c6c0-47e6-8b88-30948303e643?locale=en


![Image](https://user-images.githubusercontent.com/1054448/203325567-5308f41d-f953-4dff-8fe2-7f471569b113.png)

**Comment by orangewolf** [comment]
Rachel September 2022
---
Wow! This is great!
